spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/config/HttpClientFactory.java

```diff
@@ -117,6 +108,10 @@ protected HttpClient createInstance() {
 		return httpClient;
 	}
 
+	protected HttpClient configureSsl(HttpClient httpClient) {
+		return sslConfigurer.configureSsl(httpClient);
```

Comment:
You need to restore **all** the old `protected` methods and in the old `HttpClientFactory` constructor set `sslConfigurer` to null and here call the old methods.